main.tf

```diff
@@ -12,32 +12,46 @@ resource "helm_release" "gitlab_runner" {
   namespace        = var.namespace
   version          = var.chart_version
   create_namespace = var.create_namespace
+  atomic           = true
 
 
   values = [
-    yamlencode(merge({
+    yamlencode({
+
       image                   = var.runner_image
       gitlabUrl               = var.gitlab_url
       concurrent              = var.concurrent
       runnerRegistrationToken = var.runner_registration_token
+      replicas                = var.replicas
+      unregisterRunners       = true
+      secrets                 = var.additional_secrets
+
 
       runners = {
+        name        = var.runner_name
         runUntagged = var.run_untagged_jobs
         tags        = var.runner_tags
         locked      = var.runner_locked
         config      = local.config
+
+        cache = {
+          secretName = var.cache_secret_name
```

Comment:
Shouldn't this have some sort of derivative value?

Say 
```hcl
locals {
  cache_secret_name = var.cache_secret_name == null ? "${var.runner_name}-cache" : var.cache_secret_name
}

...

secretName = local.cache_secret_name

...
```

Reply:
cache_secretName is dependent on runners.cache.type. it has different values for GCS, AWS and Azure. and since cache_type is set to null, i thought it best to set it to null as well.

Reply:
What's the value for each of those? We can generate it based off that. And allow it to also be overwritten